### PR TITLE
Re-work Docs and split out README (using MkDocs)

### PR DESCRIPTION
**Pull request by richbeales**
### Background

The README is too long, and difficult to read/maintain.  Split into separate markdown files in the /docs folder.  Set up mkdocs to host a static site on github pages.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,477 +55,52 @@ Your support is greatly appreciated. Development of this free, open-source proje
...
-1. Run the `autogpt` python module in your terminal:
-
-    ```bash
-    python -m autogpt --speak --continuous
-    ```
-
-2. To exit the program, press Ctrl + C
-
-## GPT3.5 ONLY Mode
+## Plugins
```
Comment (ntindle) [inline README.md]: Keep plugins warning everywhere they are referenced

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Auto-GPT
```
Comment (ntindle) [inline docs/index.md:1]: not sure if this generates a side bar with a Table of Contents but if not, make one here

File: docs/plugins.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+## Plugins
+
+Use the [Auto-GPT Plugin Template](https://github.com/Significant-Gravitas/Auto-GPT-Plugin-Template) as a starting point for creating your own plugins.
```
Comment (ntindle) [inline docs/plugins.md]: drop this line, and merge the plugin changes from #2883

Comment (ntindle) [reply docs/plugins.md]: Or potentially just push these changes up to the main readme and reference them there?

File: mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+site_name: Auto-GPT
```
Comment (ntindle) [inline mkdocs.yml:1]: Add the contributing and other .md files too

**Review by ntindle — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 2 above)
Comment (richbeales) [reply docs/index.md:1]: The mkdocs site does yes.
image

File: docs/plugins.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
```
Comment (ntindle) [inline docs/plugins.md]: Drop this section or move it lower. Priority should be on using first party plugins for safety

File: docs/plugins.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+See our [Plugins Repo](https://github.com/Significant-Gravitas/Auto-GPT-Plugins) for more info on how to install all the amazing plugins the community has built!
+
+⚠️💀 **WARNING** 💀⚠️: Review the code of any plugin you use thoroughly, as plugins can execute any Python code, potentially leading to malicious activities, such as stealing your API keys.
+
+### Plugin Installation Steps
```
Comment (ntindle) [inline docs/plugins.md]: Drop this section for now or add it as wip

File: docs/plugins.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+## Plugins
+
+Use the [Auto-GPT Plugin Template](https://github.com/Significant-Gravitas/Auto-GPT-Plugin-Template) as a starting point for creating your own plugins.
+
+See our [Plugins Repo](https://github.com/Significant-Gravitas/Auto-GPT-Plugins) for more info on how to install all the amazing plugins the community has built!
+
+⚠️💀 **WARNING** 💀⚠️: Review the code of any plugin you use thoroughly, as plugins can execute any Python code, potentially leading to malicious activities, such as stealing your API keys.
```
Comment (ntindle) [inline docs/plugins.md:3]: Move this to top of segment

**Review by ntindle — COMMENTED** [review]
Much better, few changes requested

File: README.md [hunk 8]
```diff
@@ -55,477 +55,52 @@ Your support is greatly appreciated. Development of this free, open-source proje
 - 🧠 GPT-4 instances for text generation
 - 🔗 Access to popular websites and platforms
 - 🗃️ File storage and summarization with GPT-3.5
+- Extensibility with Plugins
```
Comment (Pwuts) [inline README.md]:
```suggestion
- 🔌 Extensibility with Plugins
```

File: README.md
Comment (Pwuts) [inline README.md:1]:
```md
## Installation

[...](link-to-docs)

## Usage

[...](link-to-docs)
```
to
```md
## Documentation
* [Installation](docs/installation.md)
* [Usage](docs/usage.md)
* ...
```

File: .github/workflows/documentation.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+name: Docs
+
+on:
+  push:
+    branches: [ stable ]
```
Comment (Pwuts) [inline .github/workflows/documentation.yml:5]:
```suggestion
  push:
    branches: [ stable ]
    paths:
      - 'docs/**'
      - 'mkdocs.yml'
```

File: README.md [hunk 10]
```diff
@@ -55,477 +55,52 @@ Your support is greatly appreciated. Development of this free, open-source proje
...
-**Optional:**
-- set `MILVUS_COLLECTION` in `.env` to change milvus collection name as you want, `autogpt` is the default name.
-
-
-### Weaviate Setup
-[Weaviate](https://weaviate.io/) is an open-source vector database. It allows to store data objects and vector embeddings from ML-models and scales seamlessly to billion of data objects. [An instance of Weaviate can be created locally (using Docker), on Kubernetes or using Weaviate Cloud Services](https://weaviate.io/developers/weaviate/quickstart). 
-Although still experimental, [Embedded Weaviate](https://weaviate.io/developers/weaviate/installation/embedded) is supported which allows the Auto-GPT process itself to start a Weaviate instance. To enable it, set `USE_WEAVIATE_EMBEDDED` to `True` and make sure you `pip install "weaviate-client>=3.15.4"`. 
-
-#### Install the Weaviate client
+[Speech](/docs/voice.md)
```
Comment (Torantulino) [inline README.md]: Imagegen, memory, search and voice are all throwing 404s for me.

Comment (Torantulino) [reply README.md]: I think [Imagegen, memory, search and voice] are missing the "configuration/" section from the link path.

**Review by Torantulino — DISMISSED** [review]

File: .github/workflows/documentation.yml [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    paths:
+      - 'docs/**'
+      - 'mkdocs.yml'
```
Comment (Pwuts) [inline .github/workflows/documentation.yml:8]: Also run workflow when the workflow is updated
```suggestion
    paths:
      - 'docs/**'
      - 'mkdocs.yml'
      - '.github/workflows/documentation.yml'
```

**Review by p-i- — APPROVED** [review]

**Pull request merged**